actor/cfnetworkingaction/policy.go

```diff
@@ -49,73 +52,199 @@ func (actor Actor) AddNetworkPolicy(srcSpaceGUID, srcAppName, destSpaceGUID, des
 func (actor Actor) NetworkPoliciesBySpace(spaceGUID string) ([]Policy, Warnings, error) {
 	var allWarnings Warnings
 
-	applications, warnings, err := actor.V3Actor.GetApplicationsBySpace(spaceGUID)
+	currentSpaceApps, warnings, err := actor.V3Actor.GetApplicationsBySpace(spaceGUID)
 	allWarnings = append(allWarnings, Warnings(warnings)...)
 	if err != nil {
 		return []Policy{}, allWarnings, err
 	}
 
+	var srcAppGUIDs []string
+	for _, app := range currentSpaceApps {
+		srcAppGUIDs = append(srcAppGUIDs, app.GUID)
+	}
+
 	var v1Policies []cfnetv1.Policy
-	v1Policies, err = actor.NetworkingClient.ListPolicies()
+	v1Policies, err = actor.NetworkingClient.ListPolicies(srcAppGUIDs...)
+	if err != nil {
+		return []Policy{}, allWarnings, err
+	}
+
+	v1Policies = filterPoliciesWithoutMatchingSourceGUIDs(v1Policies, srcAppGUIDs)
```

Comment:
It seems like this is doing the same thing as passing `srcAppGUIDs` as a filter on line 67 above. Why is this filtering needed?

Reply:
This filtering is to remove policies without matching source guids. The API that we are using potentially returns policies with destination app guids that match the source app guids depending on version of the networking API you are using.

Reply:
We will add a comment here to clarify that ListPolicies returns policies that have the guid in either the source or destination.